### PR TITLE
[bitnami/harbor] add ability to set serviceAccountName

### PR DESCRIPTION
**Pull request by Thomgrus**
### Description of the change

This pull request introduces the capability to specify a Kubernetes `serviceAccountName` in the deployments managed by the Harbor chart. This addition allows users to assign a specific service account to the Harbor components, enhancing integration with features like Google Workload Identity for keyless authentication to GCS storage.

### Benefits

The main benefit of this change is the flexibility it offers for Kubernetes deployments. Users can now leverage Kubernetes service accounts for fine-grained access control and integration with cloud-native features, such as keyless authentication with Google Cloud Storage. This is particularly useful for secure, seamless integration of Harbor with cloud storage solutions.

### Possible drawbacks

There are no known drawbacks to this change as it is non-breaking and transparent. By default, the behavior of the Harbor deployments remains unchanged unless a `serviceAccountName` is explicitly specified in the chart's values.

### Applicable issues

No applicable issues for now

### Additional information

This feature is a straightforward addition that aligns with the standard practices of Kubernetes deployments. It adheres to the principles of minimal disruption and backward compatibility, ensuring a smooth experience for existing users of the Harbor chart.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/harbor/README.md [hunk 1]
```diff
@@ -316,6 +316,7 @@ Additionally, if `persistence.resourcePolicy` is set to `keep`, you should manua
 | `nginx.containerSecurityContext.seccompProfile.type`      | Set container's Security Context seccomp profile                                                                         | `RuntimeDefault`        |
 | `nginx.updateStrategy.type`                               | NGINX deployment strategy type - only really applicable for deployments with RWO PVs attached                            | `RollingUpdate`         |
 | `nginx.lifecycleHooks`                                    | LifecycleHook for the NGINX container(s) to automate configuration before or after startup                               | `{}`                    |
+| `nginx.serviceAccountName`                                | Set the service account to be used by the pod, default if left empty                                                     | `""`                    |
```
Comment (fevisera) [inline bitnami/harbor/README.md:319]:
```suggestion
| `nginx.serviceAccountName`                                | Set the service account name for the NGINX pods                                                    | `""`                    |
```

File: bitnami/harbor/README.md [hunk 2]
```diff
@@ -412,6 +413,7 @@ Additionally, if `persistence.resourcePolicy` is set to `keep`, you should manua
 | `portal.initContainers`                                    | Add additional init containers to the Harbor Portal pods                                                                 | `[]`                            |
 | `portal.extraVolumeMounts`                                 | Optionally specify extra list of additional volumeMounts for the Harbor Portal pods                                      | `[]`                            |
 | `portal.extraVolumes`                                      | Optionally specify extra list of additional volumes for the Harbor Portal pods                                           | `[]`                            |
+| `portal.serviceAccountName`                                | Set the service account to be used by the pod, default if left empty                                                     | `""`                            |
```
Comment (fevisera) [inline bitnami/harbor/README.md:416]:
```suggestion
| `portal.serviceAccountName`                                | Set the service account name for the Harbor Portal pods                                                    | `""`                            |
```

File: bitnami/harbor/README.md [hunk 3]
```diff
@@ -505,6 +507,7 @@ Additionally, if `persistence.resourcePolicy` is set to `keep`, you should manua
...
 | `core.extraVolumeMounts`                                 | Optionally specify extra list of additional volumeMounts for the Harbor Core pods                                                                                                                                                                                                        | `[]`                          |
 | `core.extraVolumes`                                      | Optionally specify extra list of additional volumes for the Harbor Core pods                                                                                                                                                                                                             | `[]`                          |
+| `core.serviceAccountName`                                | Set the service account to be used by the pod, default if left empty                                                                                                                                                                                                                     | `""`                          |
```
Comment (fevisera) [inline bitnami/harbor/README.md:510]:
```suggestion
| `core.serviceAccountName`                                | Set the service account name for the Harbor Core pods                                                                                                                                                                                                                     | `""`                          |
```

File: bitnami/harbor/README.md [hunk 4]
```diff
@@ -592,6 +595,7 @@ Additionally, if `persistence.resourcePolicy` is set to `keep`, you should manua
 | `jobservice.initContainers`                                    | Add additional init containers to the Harbor Jobservice pods                                                                                                 | `[]`                                |
 | `jobservice.extraVolumeMounts`                                 | Optionally specify extra list of additional volumeMounts for the Harbor Jobservice pods                                                                      | `[]`                                |
 | `jobservice.extraVolumes`                                      | Optionally specify extra list of additional volumes for the Harbor Jobservice pods                                                                           | `[]`                                |
+| `jobservice.serviceAccountName`                                | Set the service account to be used by the pod, default if left empty                                                                                         | `""`                                |
```
Comment (fevisera) [inline bitnami/harbor/README.md:598]:
```suggestion
| `jobservice.serviceAccountName`                                | Set the service account name for the Harbor Jobservice pods                                                                                         | `""`                                |
```

File: bitnami/harbor/README.md [hunk 5]
```diff
@@ -827,6 +832,7 @@ Additionally, if `persistence.resourcePolicy` is set to `keep`, you should manua
 | `trivy.initContainers`                                    | Add additional init containers to the Trivy pods                                                                         | `[]`                                   |
 | `trivy.extraVolumeMounts`                                 | Optionally specify extra list of additional volumeMounts for the Trivy pods                                              | `[]`                                   |
 | `trivy.extraVolumes`                                      | Optionally specify extra list of additional volumes for the Trivy pods                                                   | `[]`                                   |
+| `trivy.serviceAccountName`                                | Set the service account to be used by the pod, default if left empty                                                     | `""`                                   |
```
Comment (fevisera) [inline bitnami/harbor/README.md:835]:
```suggestion
| `trivy.serviceAccountName`                                | Set the service account name for the Trivy pods                                                     | `""`                                   |
```

File: bitnami/harbor/README.md [hunk 6]
```diff
@@ -639,6 +643,7 @@ Additionally, if `persistence.resourcePolicy` is set to `keep`, you should manua
...
 | `registry.extraVolumes`                                                 | Optionally specify extra list of additional volumes for the Harbor Registry pods                                                                                                                                                                                          | `[]`                                                                                |
+| `registry.serviceAccountName`                                           | Set the service account to be used by the pod, default if left empty                                                                                                                                                                                                      | `""`                                                                                |
```
Comment (fevisera) [inline bitnami/harbor/README.md:646]:
```suggestion
| `registry.serviceAccountName`                                           | Set the service account name for the Harbor Registry pods                                                                                                                                                                                                      | `""`                                                                                |
```

**Comment by fevisera** [comment]
Hi @Thomgrus,

Thank you for your contribution. Please take a moment to review the suggestions provided in the comments.

Thanks!

**Comment by Thomgrus** [comment]
Hi @fevisera,

I've updated the Helm variable comments as you suggested. Thanks for considering my PR; let me know if there's anything else that needs to be addressed.

Thank you.

Have a nice day 👋

**Comment by fevisera** [comment]
Hi @Thomgrus,

Please, resolve the conflicts with the chart version.

Thank you!

**Comment by Thomgrus** [comment]
Hi @fevisera 👋

Conflicts resolved.

Thank you

**Review by fevisera — APPROVED** [review]
Thank you so much for your contribution!
LGTM!

**Pull request merged**